### ISSUE TITLE
Bug: Cannot perform unittest without valid key 

### ISSUE THREAD

**Issue by Domejko** [alat-rights/alana-utilities#10]
While I wanted to perform tests I did encounter this error on 12 from 16 test cases:
```
TypeError: "Could not resolve authentication method. Expected either api_key or auth_token to be set. Or for one of the `X-Api-Key` or `Authorization` headers to be explicitly omitted"
```

Test could not be performed with out Claude account with resources on it. I think that for general testing purposes we will need to mock Anthropic  `client.messages.create`.

**Comment by alat-rights** [comment]
Hi!

Yes, tests are currently run live via the Anthropic API.

While I'm open to mocking, this is not currently considered a bug.

Can you get yourself an Anthropic API key [here](https://console.anthropic.com/), and then add `os.environ["ANTHROPIC_API_KEY"] = your_api_key` to your local copy of the test code?

Definitely want to automate this with GitHub Actions in the future.

**Comment by Domejko** [comment]
I have already set up an account and used valid generated `api_key` but then it was demanding to top up my credits, there is a option go get 5$ for free but I'm not convinced to give my phone number left and right.

I'll will try to mock that return value so that testing will be independent of valid api_key.

**Comment by alat-rights** [comment]
Ah, I see!

I think a graceful way to handle this would be to keep the existing functions as-is and add new ones with mocked outputs.

Proposal:
- We can have two classes (maybe `TestFunctionsLocal` and `TestFunctionsAPI`?), and then take an argument from the terminal to decide whether to run both or to only run `TestFunctionsLocal`.

**Comment by Domejko** [comment]
Yes if you want to keep test on live API responses it can be done this way. We can also have 2 test files, one for live testing and the other for local and run them independently (but this only assuming that in future all test will be still kept in one-file format).

Choice is up to you.

**Comment by alat-rights** [comment]
Yup! Let's keep it all in `simple_tests.py` for now. We can always split into multiple files in the future.
